rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/util/FormKeyTextExtractionUtilTest.java

```diff
@@ -0,0 +1,67 @@
+package io.ona.rdt.robolectric.util;
+
+import com.vijay.jsonwizard.constants.JsonFormConstants;
+import com.vijay.jsonwizard.widgets.RepeatingGroupFactory;
+
+import org.apache.commons.lang3.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.junit.Assert;
+import org.junit.Test;
+import org.powermock.reflect.Whitebox;
+import org.smartregister.AllConstants;
+import org.smartregister.util.JsonFormUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.ona.rdt.robolectric.RobolectricTest;
+import io.ona.rdt.util.FormKeyTextExtractionUtil;
+
+public class FormKeyTextExtractionUtilTest extends RobolectricTest {
+
+    @Test
+    public void testGetFormWidgetKeyToTextMapShouldPopulateNonEmptyMap() throws Exception {
+
+        Map<String, String> expectedData = getTestFormWidgetKeyToTextMap();
+        Map<String, String> actualData = FormKeyTextExtractionUtil.getFormWidgetKeyToTextMap();
+
+        for (Map.Entry<String, String> entry : expectedData.entrySet()) {
+            String expectedValue = entry.getValue();
+            String actualValue = actualData.get(entry.getKey());
+            Assert.assertEquals(expectedValue, actualValue);
+        }
+    }
+
+    private Map<String, String> getTestFormWidgetKeyToTextMap() throws Exception {
```

Comment:
We don't need this method, the map is populated automatically and we should instead verify that the key-val pairs in the map match what we expect.

Reply:
method updated with expected key-val pair